lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -810,7 +811,7 @@ const SIMPLE_EXTRACTORS = {
 	},
 	moduleReason: {
 		_: (object, reason, { runtime }, { requestShortener }) => {
-			const depAsAny = /** @type {TODO} */ (reason.dependency);
+			const depAsAny = /** @type {ModuleDependency|undefined} */ (reason.dependency);
```

Comment:
this is legal in typescript.
https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgILIN4Chm+TYCAGwBNUAuZEAVwFsAjaLAXyy1ElkRQCFkIAHpBAkAzmkw48BYiR4B+SjQZNW7cNHhJkAYX5CII8emx58hUjqV1GUFmwQB7EKLDI4ldAF5MM0hWQARmYAbiwnFzd6Sj4fODCI12QESj04sKA

Reply:
It's legal, but not correct. The car would have the ModuleDependency, but at runtime it could also have other types. That's why it's wrong.

You must only use casts when the type is restricted at runtime but typescript doesn't know about that. Here the cast is useful to tell typescript that you are sure that this is the type here.